From cc66a01eeb8ed19bdbdc4ca23ab19c2744b2bd95 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Thu, 31 Oct 2019 15:58:24 +0530
Subject: [PATCH 399/767] octeontx2-pf: Allow setting VF mac and vlan only when
 PF is UP

commit d91c0634ae5d5e9432f054919a1952dfd91507db from
git@git.assembla.com:cavium/WindRiver.linux.git

To install NPC MCAM rules to support VF MAC and VF VLAN settings
PF needs to know ingress hw channel number. This info is sent to
PF by AF in response to NIXLF_ALLOC mbox ie NIXLF init request.

So skip 'set vf vlan' and 'set vf mac' if PF interface is not UP.

Change-Id: Ib8d0c5c84296aa22090d39038055e769668ac07f
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18079
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index cbb4b4ec7059..a425ebcbd4b8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1987,6 +1987,9 @@ static int otx2_set_vf_mac(struct net_device *netdev, int vf, u8 *mac)
 	struct pci_dev *pdev = pf->pdev;
 	struct otx2_vf_config *config;
 
+	if (!netif_running(netdev))
+		return -EAGAIN;
+
 	if (vf >= pci_num_vf(pdev))
 		return -EINVAL;
 
@@ -2034,6 +2037,9 @@ static int otx2_set_vf_vlan(struct net_device *netdev, int vf, u16 vlan, u8 qos,
 	struct pci_dev *pdev = pf->pdev;
 	struct otx2_vf_config *config;
 
+	if (!netif_running(netdev))
+		return -EAGAIN;
+
 	if (vf >= pci_num_vf(pdev))
 		return -EINVAL;
 
-- 
2.31.1

